test/CMakeLists.txt

```diff
@@ -28,75 +30,95 @@ copy_dll(finufft basicpassfail)
-  add_test(
-    NAME run_finufft1dmany_test_${PREC}
-    COMMAND finufft1dmany_test${SUFFIX} 3 1e2 2e2 ${REQ_TOL} 0 0 0 2 0.0
+    # All of the following should be run at OMP_NUM_THREADS=4 or something small,
+    # as in makefile. This prevents them taking a huge time on a, say, 128-core
+    # Rome node. ... but I don't know platform-indep way to do that! Does anyone?
+
+    add_test(
+        NAME run_basic_pass_fail_${PREC}
+        COMMAND basicpassfail${SUFFIX}
+        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    )
+
+    add_test(
+        NAME run_finufft1d_test_${PREC}
+        COMMAND finufft1d_test${SUFFIX} 1e2 2e2 ${REQ_TOL} 0 2 0.0 ${CHECK_TOL}
+        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    )
+
+    add_test(
+        NAME run_finufft1dmany_test_${PREC}
+        COMMAND
+            finufft1dmany_test${SUFFIX} 3 1e2 2e2 ${REQ_TOL} 0 0 0 2 0.0
             ${CHECK_TOL}
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
-
-  add_test(
-    NAME run_finufft2d_test_${PREC}
-    COMMAND finufft2d_test${SUFFIX} 1e2 1e1 1e3 ${REQ_TOL} 0 2 0.0 ${CHECK_TOL}
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
-
-  add_test(
-    NAME run_finufft2dmany_test_${PREC}
-    COMMAND finufft2dmany_test${SUFFIX} 3 1e2 1e1 1e3 ${REQ_TOL} 0 0 0 2 0.0
+        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    )
+
+    add_test(
+        NAME run_finufft2d_test_${PREC}
+        COMMAND
+            finufft2d_test${SUFFIX} 1e2 1e1 1e3 ${REQ_TOL} 0 2 0.0 ${CHECK_TOL}
+        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    )
+
+    add_test(
+        NAME run_finufft2dmany_test_${PREC}
+        COMMAND
+            finufft2dmany_test${SUFFIX} 3 1e2 1e1 1e3 ${REQ_TOL} 0 0 0 2 0.0
             ${CHECK_TOL}
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
-
-  add_test(
-    NAME run_finufft3d_test_${PREC}
-    COMMAND finufft3d_test${SUFFIX} 5 10 20 1e2 ${REQ_TOL} 0 2 0.0 ${CHECK_TOL}
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
-
-  add_test(
-    NAME run_finufft3dmany_test_${PREC}
-    COMMAND finufft3dmany_test${SUFFIX} 2 5 10 20 1e2 ${REQ_TOL} 0 0 0 2 0.0
+        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    )
+
+    add_test(
+        NAME run_finufft3d_test_${PREC}
+        COMMAND
+            finufft3d_test${SUFFIX} 5 10 20 1e2 ${REQ_TOL} 0 2 0.0 ${CHECK_TOL}
+        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    )
+
+    add_test(
+        NAME run_finufft3dmany_test_${PREC}
+        COMMAND
+            finufft3dmany_test${SUFFIX} 2 5 10 20 1e2 ${REQ_TOL} 0 0 0 2 0.0
             ${CHECK_TOL}
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
-
-  add_test(
-    NAME run_dumbinputs_${PREC}
-    COMMAND dumbinputs${SUFFIX}
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
-
+        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    )
+
+    add_test(
+        NAME run_dumbinputs_${PREC}
+        COMMAND dumbinputs${SUFFIX}
+        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    )
+
+    add_test(
+        NAME spreadinterp1d_${PREC}
+        COMMAND spreadinterp1d_test${SUFFIX} 1e6 1e6 1e-5 1 2 2.0 1e-4
```

Comment:
Thanks for adding this CPU test. But this is failing since you need to respect the tolerances.    Try 1e3 1e3 ${REQ_TOL} 1 2 2.0 ${CHECK_TOL}